src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,140 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
```

Comment:
Consider making `Node<T>` an inner class of `MyLinkedList` to encapsulate it within `MyLinkedList` as it is specific to your linked list implementation.